src/ctapipe_io_lst/calibration.py

```diff
@@ -356,35 +364,77 @@ def _read_calibration_file(path):
         Read the correction from hdf5 calibration file
         """
 
-        with tables.open_file(path) as f:
-            tel_groups = [
-                key for key in f.root._v_children.keys()
-                if key.startswith('tel_')
-            ]
-
         mon = MonitoringContainer()
 
-        for base in tel_groups:
-            with HDF5TableReader(path) as h5_table:
-                # read the calibration data
-                tel_id = int(base[4:])
-                mon.tel[tel_id] = MonitoringCameraContainer(
-                    calibration=next(h5_table.read(f'/{base}/calibration', WaveformCalibrationContainer)),
-                    pedestal=next(h5_table.read(f'/{base}/pedestal', PedestalContainer)),
-                    flatfield=next(h5_table.read(f'/{base}/flatfield', FlatFieldContainer)),
-                    pixel_status=next(h5_table.read(f"/{base}/pixel_status", PixelStatusContainer)),
-                )
+        if path.name.endswith(".h5"):
+            # get keys in file
+            with tables.open_file(path) as f:
+                tel_groups = [
+                    key for key in f.root._v_children.keys()
+                    if key.startswith('tel_')
+                ] 
+
+            for base in tel_groups:
+                with HDF5TableReader(path) as h5_table:
+                    # read the calibration data
+                    tel_id = int(base[4:])
+                    mon.tel[tel_id] = MonitoringCameraContainer(
+                        calibration=next(h5_table.read(f'/{base}/calibration', WaveformCalibrationContainer)),
+                        pedestal=next(h5_table.read(f'/{base}/pedestal', PedestalContainer)),
+                        flatfield=next(h5_table.read(f'/{base}/flatfield', FlatFieldContainer)),
+                        pixel_status=next(h5_table.read(f"/{base}/pixel_status", PixelStatusContainer)),
+                    )
+        
+        elif path.name.endswith(".fits") or path.name.endswith(".fits.gz"):
+            
+            CALIB_CONTAINERS = {
+                "calibration": WaveformCalibrationContainer,
+                "flatfield": FlatFieldContainer,
+                "pedestal": PedestalContainer,
+                "pixel_status": PixelStatusContainer,
+            } 
+
+            mon_data = MonitoringCameraContainer()
+         
+            with fits.open(path) as f:
+                tel_id = f['PRIMARY'].header['TEL_ID']
+                for key in CALIB_CONTAINERS.keys():
+                    table = QTable.read(f, hdu=key)
+                    row = table[0]
+                    
+                    for col in row.keys():  
+                        mon_data[key][col] = row[col]
+            
+            # fits forget units if not defined in the container (to be corrected in ctapipe)
+            mon_data.calibration.time_correction = mon_data.calibration.time_correction*u.ns
```

Comment:
What has to be corrected in ctapipe? You are not using any feature of ctapipe here

Reply:
yes, the ctapipe containe when writing, it has not units defined

Reply:
That doesn't matter here. You are reading from FITS.

Reply:
no, as far as I understand the problem is  [here](https://github.com/cta-observatory/ctapipe/blob/7ef60cc5331f1880adccca93af26fbfd6b6c4ce8/src/ctapipe/containers.py#L1348 ), for fields with the units in the container (as [here](https://github.com/cta-observatory/ctapipe/blob/7ef60cc5331f1880adccca93af26fbfd6b6c4ce8/src/ctapipe/containers.py#L1330)) the unit is correctly propagated

Reply:
I will repeat myself, but: This branch is where we read a fits file written by lstcam_calib. If the unit is not there, it's our fault. Not ctapipe.